### DISCUSSION TITLE
C# 11 || `[CollectionDefinition<MyCollectionFixtureType>]` Generic `CollectionDefinitionAttribute` functionality?

### DISCUSSION THREAD

**Discussion by jeffward01** [xunit/xunit#2627] [General]
I reviewed the code and I do not think this feature exists, but I just wanted to ask anyways.

# Question

Is it possible to use `[CollectionDefinition("myCollectionFixtureName")]` with generic attributes?

[Here is the current documentation ](https://xunit.net/docs/shared-context)for `ICollectionFixture<>`

### Current 

```csharp

[CollectionDefinition("Database collection")]
public class DatabaseCollection : ICollectionFixture<DatabaseFixture>
{
    // This class has no code, and is never created. Its purpose is simply
    // to be the place to apply [CollectionDefinition] and all the
    // ICollectionFixture<> interfaces.
}



[Collection("Database collection")]
public class DatabaseTestClass1
{
    DatabaseFixture fixture;

    public DatabaseTestClass1(DatabaseFixture fixture)
    {
        this.fixture = fixture;
    }
}

````

### Expected C# 11 functionality

```csharp

[CollectionDefinition<DatabaseCollection>]
public class DatabaseCollection : ICollectionFixture<DatabaseFixture>
{
    // This class has no code, and is never created. Its purpose is simply
    // to be the place to apply [CollectionDefinition] and all the
    // ICollectionFixture<> interfaces.
}



[Collection<DatabaseCollection>]
public class DatabaseTestClass1
{
    DatabaseFixture fixture;

    public DatabaseTestClass1(DatabaseFixture fixture)
    {
        this.fixture = fixture;
    }
}

```

-----

## Summary
* Is this possible?
*  Is this something that I can open an issue for and myself _(alas... if i have time)_ or someone else can  implement?

**Comment by bradwilson** [comment]
It is not possible today.

I would have to see how this is implemented to know whether we could do it without making C# 11 the minimum language version for xUnit.net. I will convert this to an issue for you.

#2650
